pkg/resource/deploy/source_eval.go

```diff
@@ -2147,30 +2155,29 @@ func downgradeOutputValues(v resource.PropertyMap) resource.PropertyMap {
 	return result
 }
 
-func addOutputDependencies(deps []resource.URN, v resource.PropertyValue) []resource.URN {
+func addOutputDependencies(deps mapset.Set[resource.URN], v resource.PropertyValue) {
```

Comment:
At every external callsite the set is constructed from a slice, then addOutputDependencies is called on it, and then the set is converted back to a slice.

Would it be cleaner to have a wrapper function for `addOutputDependencies` that took a slice of URNS, and a propertyValue and returned a slice of URNs, then have that contain the pattern. 

something like

``` go
func extendOutputDependencies(existing []resource.URN, v resource.PropertyValue) []resource.URN {
    deps := mapset.NewSet(existing...)
    addOutputDependencies(deps, v)
    return deps.ToSlice()
}
```

Reply:
Maybe, I'll recheck how this interacts with the other changes I'm doing that use the sets even more.

Reply:
Yeh this looks reasonable, I'll add this